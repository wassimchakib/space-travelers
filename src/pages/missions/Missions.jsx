import Missions from '../../components/missions';

const MissionsPage = () => (
  <>
    <Missions />
  </>
);

export default MissionsPage;
