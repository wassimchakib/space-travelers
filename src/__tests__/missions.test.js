import { useSelector } from 'react-redux';
import { render } from '@testing-library/react';
import Mission from '../components/missions/mission';
import Missions from '../components/missions';
import JoinMissionButton from '../components/joinMissionButton';

jest.mock('react-redux');

describe('Test suite for missions components', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('should render mission component', () => {
    const tree = render(
      <table>
        <tbody>
          <Mission
            missionName="Thaicom"
            joinedMission={false}
            missionId="9D1B7E0"
            description="Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology."
          />
        </tbody>
      </table>
    );

    const title = tree.getByText('Thaicom');

    expect(tree).toMatchSnapshot();

    expect(title.textContent).toBe('Thaicom');
  });

  test('should render missions component', () => {
    useSelector.mockReturnValue([
      { missionId: '9D1B7E0', missionName: 'Thaicom', description: 'test', joinedMission: false },
    ]);

    const tree = render(<Missions />);

    expect(tree).toMatchSnapshot();
  });

  test('should render joinMissionButton with joinedMission equal to false ', () => {
    const tree = render(<JoinMissionButton joinedMission={false} missionId="9D1B7E0" />);
    expect(tree).toMatchSnapshot();
  });

  test('should render joinMissionButton with joinedMission equal to true ', () => {
    const tree = render(<JoinMissionButton joinedMission missionId="9D1B7E0" />);
    expect(tree).toMatchSnapshot();
  });
});
