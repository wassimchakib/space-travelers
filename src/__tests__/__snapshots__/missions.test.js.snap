// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for missions components should render joinMissionButton with joinedMission equal to false  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test suite for missions components should render joinMissionButton with joinedMission equal to true  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test suite for missions components should render mission component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table>
        <tbody>
          <tr>
            <td
              class="td title-container"
            >
              <h1
                class="mission-title"
              >
                Thaicom
              </h1>
            </td>
            <td
              class="td"
            >
              Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
            </td>
            <td
              class="td td-with-span"
            >
              <span
                class="not-a-member status"
              >
                NOT A MEMBER
              </span>
            </td>
            <td
              class="td button-td"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table>
      <tbody>
        <tr>
          <td
            class="td title-container"
          >
            <h1
              class="mission-title"
            >
              Thaicom
            </h1>
          </td>
          <td
            class="td"
          >
            Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
          </td>
          <td
            class="td td-with-span"
          >
            <span
              class="not-a-member status"
            >
              NOT A MEMBER
            </span>
          </td>
          <td
            class="td button-td"
          />
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test suite for missions components should render missions component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="container"
        >
          <table
            class="table-container"
          >
            <thead
              id="thead"
            >
              <tr>
                <th
                  class="th"
                >
                  Missions
                </th>
                <th
                  class="th"
                >
                  Description
                </th>
                <th
                  class="th"
                >
                  Status
                </th>
                <th
                  aria-label="empty-table-head"
                  class="th"
                />
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  class="td title-container"
                >
                  <h1
                    class="mission-title"
                  >
                    Thaicom
                  </h1>
                </td>
                <td
                  class="td"
                >
                  test
                </td>
                <td
                  class="td td-with-span"
                >
                  <span
                    class="not-a-member status"
                  >
                    NOT A MEMBER
                  </span>
                </td>
                <td
                  class="td button-td"
                />
              </tr>
            </tbody>
          </table>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="container"
      >
        <table
          class="table-container"
        >
          <thead
            id="thead"
          >
            <tr>
              <th
                class="th"
              >
                Missions
              </th>
              <th
                class="th"
              >
                Description
              </th>
              <th
                class="th"
              >
                Status
              </th>
              <th
                aria-label="empty-table-head"
                class="th"
              />
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="td title-container"
              >
                <h1
                  class="mission-title"
                >
                  Thaicom
                </h1>
              </td>
              <td
                class="td"
              >
                test
              </td>
              <td
                class="td td-with-span"
              >
                <span
                  class="not-a-member status"
                >
                  NOT A MEMBER
                </span>
              </td>
              <td
                class="td button-td"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
